api/project.rb

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env ruby
+
+unless Dir.exists? ".project/common"
+  unless system(*%W{mkdir -p .project})
+    STDERR.puts "mkdir failed."
+    exit 1
+  end
+  unless system(*%W{git clone https://github.com/dmohs/project-management.git .project/common})
+    STDERR.puts "git clone failed."
+    exit 1
+  end
+end
+
+require_relative ".project/common/common"
+Dir.foreach(".project") do |item|
+  unless item =~ /^\.\.?$/ || item == "common"
```

Comment:
This searches the `.project` folder and loads everything that isn't `.`, `..`, or `common` (the source repo). That lets you dump a new script into that directory and it will be exposed in `./project.rb` without explicitly loading it. Your question makes me think this was a poor tradeoff, especially because these developer scripts change much less frequently than other parts of the repo.